src/python/pants/backend/awslambda/python/rules.py

```diff
@@ -54,6 +55,17 @@ class PythonAwsLambdaFieldSet(PackageFieldSet):
 async def package_python_awslambda(
     field_set: PythonAwsLambdaFieldSet, lambdex: Lambdex, union_membership: UnionMembership
 ) -> BuiltPackage:
+
+    if Platform.is_macos:
```

Comment:
`Platform.is_macos` is a property, not a class property.

```suggestion
async def package_python_awslambda(
    field_set: PythonAwsLambdaFieldSet,
    lambdex: Lambdex,
    platform: Platform,
    union_membership: UnionMembership
) -> BuiltPackage:
    if platform.is_macos:
```